backend/danswer/server/documents/models.py

```diff
@@ -201,11 +224,23 @@ class CCPairFullInfo(BaseModel):
     def from_models(
         cls,
         cc_pair_model: ConnectorCredentialPair,
-        index_attempt_models: list[IndexAttempt],
         latest_deletion_attempt: DeletionAttemptSnapshot | None,
+        number_of_index_attempts: int,
+        last_index_attempt: IndexAttempt | None,
         num_docs_indexed: int,  # not ideal, but this must be computed separately
         is_editable_for_current_user: bool,
     ) -> "CCPairFullInfo":
+        # figure out if we need to artificially deflate the number of docs indexed.
+        # This is required since the total number of docs indexed by a CC Pair is
+        # updated before the new docs for an indexing attempt. If we don't do this,
+        # there is a mismatch between these two numbers which may confuse users.
+        last_indexing_status = last_index_attempt.status if last_index_attempt else None
+        if (
+            last_indexing_status == IndexingStatus.SUCCESS
+            and number_of_index_attempts == 1
+        ):
+            num_docs_indexed = last_index_attempt.new_docs_indexed
+
```

Comment:
we now return the last indexing status and the number of docs to facilitate some of the original display logic on the frontend without having to send back the indexattempts themselves